libs/constrained-generators/src/Constrained/List.hs

```diff
@@ -113,6 +126,17 @@ fromWholeCtx :: ListCtxWhole f as c -> ListCtx f as c
 fromWholeCtx (ListCtxWhole Nil hole suf) = hole :? suf
 fromWholeCtx (ListCtxWhole (x :> pre) hole suf) = x :! fromWholeCtx (ListCtxWhole pre hole suf)
 
+fillListCtx :: ListCtx f as c -> (forall a. c a -> f a) -> List f as
+fillListCtx (ListCtx pre c post) f = appendList pre (f c :> post)
+
+mapListCtx :: (forall a. f a -> g a) -> ListCtx f as c -> ListCtx g as c
+mapListCtx nt (ListCtx pre c post) = ListCtx (mapList nt pre) c (mapList nt post)
```

Comment:
I wonder if it's possible to define this in terms of `mapListCtxC`, since it looks like a more constrained version of that function.

Reply:
Annoyingly, not quite (see the duplication in the other functions of similar type). The problem is that this would require `All SomeTrivialConstraint as` as a precondition to `mapListCtx`. Unfortunately (for good reason) you can't write `All (\_ -> ()) as` and have it not pollute everything.

Reply:
Your intuition is right, but GHCs hamstrung dependent types make it impossible in practice.